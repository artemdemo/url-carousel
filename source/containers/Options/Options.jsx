import React, { Component } from 'react';
import Form from '../../components/Form/Form';
import Input from '../../components/Input/Input';
import ContentBlock from '../../components/ContentBlock/ContentBlock';
import UrlList from '../UrlList/UrlList';

import './Options.less';

class Options extends Component {
    constructor(props) {
        super(props);
        this.state = {
            url: '',
            urlError: false,
        };
    }

    submit() {
        if (this.state.url === '') {
            this.setState({urlError: 'Url should not be empty'});
            return false;
        }
    }

    render() {
        return (
            <div className='options'>
                <Form onSubmit={() => this.submit()}>
                    <ContentBlock>
                        <div className='row'>
                            <div className='col-xs-9'>
                                <Input
                                    value={this.state.url}
                                    onChange={e => this.setState({
                                        url: e.target.value,
                                        urlError: false,
                                    })}
                                    placeholder='Url' />
                            </div>
                            <div className='col-xs-3'>
                                <button
                                    className='btn btn-default'>Add</button>
                            </div>
                        </div>
                    </ContentBlock>
                </Form>
                <UrlList />
            </div>
        );
    }
}

export default Options;
