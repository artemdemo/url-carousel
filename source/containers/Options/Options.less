.options {
    padding: 0 10px 5px 10px;
}
