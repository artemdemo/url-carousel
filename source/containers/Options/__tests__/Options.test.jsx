import React from 'react';
import renderer from 'react-test-renderer';
import Options from '../Options';

describe('<Options>', () => {
    it('Render', () => {
        const tree = renderer.create(
            <Options />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
