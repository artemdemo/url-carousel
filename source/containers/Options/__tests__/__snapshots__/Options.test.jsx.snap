// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options> Render 1`] = `
<div
  className="options"
>
  <form
    className={undefined}
    onSubmit={[Function]}
  >
    <div
      className="content-block"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Url"
        value=""
      />
    </div>
    <div
      className="content-block"
    >
      <button
        className="btn btn-default"
      >
        Add
      </button>
    </div>
  </form>
  <div>
    <h4>
      URL List
    </h4>
    <div>
      http://www.facebook.com
    </div>
  </div>
</div>
`;
